recipes/civetweb/all/conanfile.py

```diff
@@ -95,53 +88,56 @@ def validate(self):
             raise ConanInvalidConfiguration("ssl_dynamic_loading requires shared openssl")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
 
         if self.options.with_ssl:
-            openssl_version = tools.Version(self.deps_cpp_info["openssl"].version[:-1])
-            self._cmake.definitions["CIVETWEB_ENABLE_SSL"] = self.options.with_ssl
-            self._cmake.definitions["CIVETWEB_ENABLE_SSL_DYNAMIC_LOADING"] = self.options.ssl_dynamic_loading
-            self._cmake.definitions["CIVETWEB_SSL_OPENSSL_API_1_0"] = openssl_version.minor == "0"
-            self._cmake.definitions["CIVETWEB_SSL_OPENSSL_API_1_1"] = openssl_version.minor == "1"
-
-        self._cmake.definitions["CIVETWEB_BUILD_TESTING"] = False
-        self._cmake.definitions["CIVETWEB_CXX_ENABLE_LTO"] = False
-        self._cmake.definitions["CIVETWEB_ENABLE_ASAN"] = False
-
-        self._cmake.definitions["CIVETWEB_DISABLE_CACHING"] = not self.options.with_caching
-        self._cmake.definitions["CIVETWEB_DISABLE_CGI"] = not self.options.with_cgi
-        self._cmake.definitions["CIVETWEB_ENABLE_CXX"] = self.options.with_cxx
-        self._cmake.definitions["CIVETWEB_ENABLE_DUKTAPE"] = self.options.with_duktape
-        self._cmake.definitions["CIVETWEB_ENABLE_IPV6"] = self.options.with_ipv6
-        self._cmake.definitions["CIVETWEB_ENABLE_LUA"] = self.options.with_lua
-        self._cmake.definitions["CIVETWEB_ENABLE_SERVER_STATS"] = self.options.with_server_stats
-        self._cmake.definitions["CIVETWEB_ENABLE_THIRD_PARTY_OUTPUT"] = self.options.with_third_party_output
-        self._cmake.definitions["CIVETWEB_ENABLE_WEBSOCKETS"] = self.options.with_websockets
-        self._cmake.definitions["CIVETWEB_SERVE_NO_FILES"] = not self.options.with_static_files
+            openssl_version = Version(self.deps_cpp_info["openssl"].version[:-1])
```

Comment:
```suggestion
            openssl_version = Version(self.depencies["openssl"].ref.version[:-1])
```
https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface